pvlib/clearsky.py

```diff
@@ -635,16 +654,26 @@ def _clear_sample_index(clear_windows, samples_per_window, align, H):
     #     shift = - (samples_per_window // 2)
     # else:
     #     shift = 0
-    shift = -(samples_per_window // 2)
-    idx = clear_windows.shift(shift)
+
+    # Account for the row # on which the interval is centered not actually
+    # being in row samples_per_window // 2 if samples_per_window is even
+    if samples_per_window % 2 == 0:
+        shift = -(samples_per_window // 2 - 1)
+    else:
+        shift = -(samples_per_window // 2)
+    clear_cols = clear_windows.shift(shift)
     # drop rows at the end corresponding to windows past the end of data
-    idx = idx.drop(clear_windows.index[1 - samples_per_window:])
-    idx = idx.astype(bool)  # shift changed type to object
-    clear_samples = np.unique(H[:, idx])
+    clear_cols = clear_cols.drop(clear_windows.index[1 - samples_per_window:])
+    clear_cols = clear_cols.astype(bool)  # shift changed type to object
+    # Boolean mask for column indices of intervals with temporal gaps
+    gap_cols = [True if c not in gaps else False for c in range(0,
+                len(clear_windows) - (samples_per_window - 1))]
+    mask = np.logical_and(clear_cols, gap_cols)
+    clear_samples = np.unique(H[:, mask])
     return clear_samples
 
 
-def detect_clearsky(measured, clearsky, times=None, window_length=10,
+def detect_clearsky(measured, clear_sky, times=None, window_length=10,
```

Comment:
Is the arg name change `clearsky` to `clear_sky` intentional? If we keep this, need to modify the docstring.

Reply:
I was trying to avoid overloading the clearsky namespace in my development environment, as I also had clearsky defined with `from pvlib import clearsky`. If this change is okay with the maintainers, I will modify the docstring.

Reply:
I think we want to avoid changing the arg name here.

Reply:
Reverted to original name